src/Maestro/Maestro.Contracts/MergePolicyEvalutationResult.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Maestro.Contracts
+{
+    public class MergePolicyEvaluationResults
+    {
+        public MergePolicyEvaluationResults(IEnumerable<MergePolicyEvaluationResult> results)
+        {
+            Results = results.ToImmutableList();
+        }
+
+        public IReadOnlyList<MergePolicyEvaluationResult> Results { get; }
+
+        public bool Succeeded => Results.Count > 0 && Results.All(r => r.Status == MergePolicyEvaluationStatus.Success);
+
+        public bool Pending => Results.Count > 0 && Results.Any(r => r.Status == MergePolicyEvaluationStatus.Pending);
+
+        public bool Failed => Results.Count > 0 && Results.Any(r => r.Status == MergePolicyEvaluationStatus.Failure);
+    }
+
+    public class MergePolicyEvaluationResult
+    {
+        public MergePolicyEvaluationResult(MergePolicyEvaluationStatus status, string message, IMergePolicyInfo mergePolicy)
+        {
+            if (mergePolicy.Name == null)
+            {
+                throw new ArgumentNullException(nameof(mergePolicy.Name));
```

Comment:
This is going to produce a weird exception, because it's just going to say that the "Name" argument is null... but this method doesn't have a "Name" parameter.  It should probably just be an ArgumentException with some text that explains it.

It's also odd that they _actual_ argument, "mergePolicy" isn't checked, and will just crash with a NullReferenceException.